tenstorrent/tt-metal
# Added register API + noc tests

- Added register read API 
- tests 
    - verify each NOC node read from device matches the worker node xy we mapped to in software. 
    - verify bottom part of translation table is identity

fixes #2844, #2843
